src/tasks/test-deployment.ts

```diff
@@ -0,0 +1,390 @@
+      !Object.keys(defaultByChainId).includes(chainId)
+    ) {
+      throw new Error(
+        `Chain id ${chainId} does not have a default address for ${token}`,
+      );
+    }
+    const defaultAddress =
+      defaultByChainId[chainId as keyof typeof defaultByChainId];
+    return address ?? defaultAddress;
+  }
+  async function getToken(address: string): Promise<Token> {
+    const instance = new Contract(address, IERC20.abi).connect(ethers.provider);
+    const decimals = await instance.decimals();
+    if (typeof decimals !== "number") {
+      throw new Error(
+        `Invalid number of decimals for token at address ${address}`,
+      );
+    }
+    return {
+      instance,
+      decimals,
+    };
+  }
+  const [usdc, gno, weth] = await Promise.all([
+    getToken(defaultIfUnset(args.usdcToken, "usdc")),
+    getToken(defaultIfUnset(args.gnoToken, "gno")),
+    getToken(defaultIfUnset(args.wethToken, "weth")),
+  ]);
+  return {
+    chainId,
+    mnemonic: args.mnemonic,
+    userCount: args.userCount ?? defaultArgs.userCount,
+    totalSupply: utils.parseUnits(
+      args.totalSupply ?? defaultArgs.totalSupply,
+      metadata.real.decimals,
+    ),
+    usdc,
+    usdcPerCow: utils.parseUnits(
+      args.usdcPerCow ?? defaultArgs.usdcPerCow,
+      usdc.decimals,
+    ),
+    gno,
+    usdcPerGno: utils.parseUnits(
+      args.usdcPerGno ?? defaultArgs.usdcPerGno,
+      usdc.decimals,
+    ),
+    weth,
+    usdcPerWeth: utils.parseUnits(
+      args.usdcPerWeth ?? defaultArgs.usdcPerWeth,
+      usdc.decimals,
+    ),
+  };
+}
+
+const setupTestDeploymentTask: () => void = () => {
+  task(
+    "test-deployment",
+    "Generate a list of pseudorandom claims for each signer and deploy test contracts on the current network.",
+  )
+    .addParam("mnemonic", "The mnemonic used to generate user addresses.")
+    .addOptionalParam(
+      "userCount",
+      "Random claims will be generated for this amount of users. Their secret key will be generated from the mnemonic.",
+      defaultArgs.userCount,
+      types.int,
+    )
+    .addOptionalParam(
+      "totalSupply",
+      "The total supply of real token minted on deployment.",
+      defaultArgs.totalSupply,
+      types.string,
+    )
+    .addOptionalParam("usdcToken", "Address of token USDC.")
+    .addOptionalParam("gnoToken", "Address of token GNO.")
+    .addOptionalParam("wethToken", "Address of token WETH.")
+    .addOptionalParam(
+      "usdcPerCow",
+      "How many USDC a COW is worth.",
+      defaultArgs.usdcPerCow,
+      types.string,
+    )
+    .addOptionalParam(
+      "usdcPerGno",
+      "How many USDC a GNO is worth.",
+      defaultArgs.usdcPerGno,
+      types.string,
+    )
+    .addOptionalParam(
+      "usdcPerWeth",
+      "How many USDC a WETH is worth.",
+      defaultArgs.usdcPerWeth,
+      types.string,
+    )
+    .setAction(async (args, hre) => {
+      await generateClaimsAndDeploy(await parseArgs(args, hre), hre);
```

Comment:
Maybe it would have been easer to digest, if you are calling here 3 different functions:
```
await generateClaims - builds the claims
await deploySetup -> does the deployment
await documentSetup -> writes all files
```
This would also make it easier to swtich the generateClaims function,( that is our next task).

Reply:
Since this is already a very heavy PR, I'll merge it as it is and split the two in a separate PR.